CHANGELOG.md

```diff
@@ -2,6 +2,22 @@
 
 This changelog will be used to generate documentation on [release notes page](http://azure.microsoft.com/en-us/documentation/articles/app-insights-release-notes-dotnet/).
 
+## Version 2.2.0-beta3
+
+- Read InstrumentationKey from environment variable APPINSIGHTS_INSTRUMENTATIONKEY if it is was not provided inline. If provided it overrides what is set though configuration file. (Feature is not available in PCL version of SDK).
+- Context properties `NetworkType`, `ScreenResolution` and `Language` marked as obsolete. Please use custom properties to report network type, screen resolution and language. Values stored in these properties will be send as custom properties. 
+- Dependency type was updated to reflect the latest developments in Applicaiton Insights Applicaiton Map feature. You can set a new field - `Target`. `CommandName` was renamed to `Data` for consistancy with the Application Analytics schema. `DependencyKind` will never be send any more and will not be set to "Other" by default. Also there are two more constructors for `DependencyTelemetry` item.
+- Type `SessionStateTelemetry` was marked obsolete. Use `IsFirst` flag in `SessionContext` to indicate that the session is just started.
+- Type `PerformanceCounterTelemetry` was marked obsolete. Use `MetricTelemetry` instead.
+- Marked `RequestTelemetry.HttpMethod` as obsolete. Put http verb as part of the name for the better grouping by name and use custom properties to report http verb as a dimension.
+- Marked `RequestTelemetry.StartTime` as obsolete. Use `TimeStamp` instead.
+- [Removed BCL dependency](https://github.com/Microsoft/ApplicationInsights-dotnet/issues/175)
+- [Added IPv6 support](https://github.com/Microsoft/ApplicationInsights-dotnet/issues/316)
+- [Fixed issue #278](https://github.com/Microsoft/ApplicationInsights-dotnet/issues/278)
```

Comment:
changelog with the issues numbers only sucks! Let's change them to something human-friendly